core/use-swr.ts

```diff
@@ -588,10 +588,15 @@ export const useSWRHandler = <Data = any, Error = any>(
     let timer: any
 
     function next() {
+      const cacheData = getCache().data
+      const configFallbackData = getConfig().fallbackData
+      const latestFallback = isUndefined(configFallbackData)
+        ? getConfig().fallback[key]
+        : configFallbackData
       // Use the passed interval
       // ...or invoke the function with the updated data to get the interval
       const interval = isFunction(refreshInterval)
-        ? refreshInterval(data)
+        ? refreshInterval(isUndefined(cacheData) ? latestFallback : cacheData)
```

Comment:
The original purpose of this feature is to change the refreshInterval based on **updated fetched data**
ref: #1686

It would be better if we only use cached data here to keep it simple.

Reply:
@promer94 

> The original purpose of this feature is to change the refreshInterval based on updated fetched data
> ref: #1686

You might put the wrong issue number, and #1686 is not relevant to this.

> It would be better if we only use cached data here to keep it simple.

Yeah, I agree with you, and it would be good if we could use cache data only.
But the current implementation uses fallback data as the `data`, so it's safer to keep the behavior to avoid a breaking change. It makes sense for me to pass fallback data when cache data is `undefined`. We could have the change in the next major release.

We might have the change as a bug fix, but I'm not sure how the change affects users.

Reply:
Thanks. i just updated the wrong issue number.

Since this feature actually never works as expected after it got released. I feel it is safe to change the implementation in a minor version here. User can still access `fallbackData` from clouser. I guess It won't be a big problem here ?